src/Simple.php

```diff
@@ -637,14 +637,24 @@ public function process($serial)
             // committed successfully (if single transaction mode has been activated)
             $this->getEmitter()->emit(EventNames::APP_PROCESS_TRANSACTION_SUCCESS, $this);
         } catch (ApplicationFinishedException $afe) {
+            // rollback the transaction, if single transaction mode has been configured
+            if ($this->getConfiguration()->isSingleTransaction()) {
+                $this->getImportProcessor()->getConnection()->rollBack();
+            }
+
+            // invoke the event that has to be fired after the application rollbacked the
+            // transaction (if single transaction mode has been activated)
+            $this->getEmitter()->emit(EventNames::APP_PROCESS_TRANSACTION_FAILURE, $this, $afe);
```

Comment:
in my understanding we do not have to rollback a transaction, because we finish without errors and don't really abort stuff... in the end, it does not make a difference because in our case we don't write anything to the database

Reply:
This will be started automatically when the CLI option `--single-transaction=true` so we should make sure it'll be rolled back, I think.